backend/src/chain/chain.py

```diff
@@ -37,15 +36,15 @@ def __init__(self, node_id: NodeId, schema_id: str):
         self.parent: None = None
         self.__node = None
 
-    def get_node(self) -> IteratorNodeBase:
+    def get_node(self) -> NodeData:
         if self.__node is None:
-            node = NodeFactory.get_node(self.schema_id)
-            assert isinstance(node, IteratorNodeBase), "Invalid iterator node"
+            node = registry.get_node(self.schema_id)
+            assert node.type == "iterator", "Invalid iterator node"
             self.__node = node
         return self.__node
 
 
-Node = Union[FunctionNode, IteratorNode]
+Nodes = Union[Node, IteratorNode]
```

Comment:
"Nodes" implies that this type is some sort of collection, which it is absolutely not. Just keep the old names as `Node` and `FunctionNode`.

Reply:
My bad, i thought FunctionNode was something i did and that your comment from earlier meant to completely change it